apps/src/gamelab/GameLabJr.interpreted.js

```diff
@@ -256,68 +256,71 @@ function hideTitleScreen() {
 function draw() {
   background(World.background_color || "white");
 
-  // Run input events
-  for (var i=0; i<inputEvents.length; i++) {
-    const eventType = inputEvents[i].type;
-    const event = inputEvents[i].event;
-    const param = inputEvents[i].param;
-    if (eventType(param)) {
-      event();
+  if (World.frameCount > 1) {
+    // Run input events
+    for (var i=0; i<inputEvents.length; i++) {
+      const eventType = inputEvents[i].type;
+      const event = inputEvents[i].event;
+      const param = inputEvents[i].param;
+      if (eventType(param)) {
+        event();
+      }
     }
-  }
 
 
-  // Run collision events
-  for (let i=0; i<collisionEvents.length; i++) {
-    const collisionEvent = collisionEvents[i];
-    const a = collisionEvent.a;
-    const b = collisionEvent.b;
-    if (a.overlap(b)) {
-      if (!collisionEvent.touching || collisionEvent.keepFiring) {
-        collisionEvent.event();
+    // Run collision events
+    for (let i=0; i<collisionEvents.length; i++) {
+      const collisionEvent = collisionEvents[i];
+      const a = collisionEvent.a;
+      const b = collisionEvent.b;
+      if (a.overlap(b)) {
+        if (!collisionEvent.touching || collisionEvent.keepFiring) {
+          collisionEvent.event();
+        }
+        collisionEvent.touching = true;
+      } else {
+        collisionEvent.touching = false;
       }
-      collisionEvent.touching = true;
-    } else {
-      collisionEvent.touching = false;
     }
-  }
 
-  // Run loops
-  for (let i=0; i<loops.length; i++) {
-    var loop = loops[i];
-    if (!loop.condition()) {
-      loops.splice(i, 1);
-    } else {
-      loop.loop();
+    // Run loops
+    for (let i=0; i<loops.length; i++) {
+      var loop = loops[i];
+      if (!loop.condition()) {
+        loops.splice(i, 1);
+      } else {
+        loop.loop();
+      }
     }
-  }
 
-  drawSprites();
 
-  for (let i=0; i<sprites.length; i++) {
-    var sprite = sprites[i];
+    for (let i=0; i<sprites.length; i++) {
+      var sprite = sprites[i];
 
-    // Perform sprite behaviors
+      // Perform sprite behaviors
 
-    for (var j=0; j<sprite.behavior_keys.length; j++) {
-      sprite.behaviors[sprite.behavior_keys[j]](sprite);
-    }
+      for (var j=0; j<sprite.behavior_keys.length; j++) {
+        sprite.behaviors[sprite.behavior_keys[j]](sprite);
+      }
 
-    // Make sprites say things
-    if (sprite.things_to_say.length > 0) {
-      fill("white");
-      rect(sprite.x + 10, sprite.y - 15, sprite.things_to_say[0][0].length * 7, 20);
-      fill("black");
-      text(sprite.things_to_say[0][0], sprite.x + 15, sprite.y);
-
-      if (sprite.things_to_say[0][1] === 0) {
-        sprite.things_to_say.shift();
-      } else if (sprite.things_to_say[0][1] > -1) {
-        sprite.things_to_say[0][1]--;
+      // Make sprites say things
```

Comment:
[Out of scope for this PR] Still used in the curriculum?

Reply:
As far as I've seen, it's not in any of our planned lessons. I imagine we'll still want to support it for freeplay, with at least the same level of personal info filtering we have for playlab.